libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -6084,3 +6081,120 @@ nubOrdMemberSpec message xs =
         , "The input is the empty list."
         ]
     )
+
+-- ==============================================================================
+-- Experimental changes to conformsToSpecM
+
+runTermE :: Env -> Term fn a -> Either (NE.NonEmpty String) a
+runTermE env = \case
+  Lit a -> Right a
+  V v -> case lookupEnv env v of
+    Just a -> Right a
+    Nothing -> Left (pure ("Couldn't find " ++ show v ++ " in " ++ show env))
+  App f ts -> do
+    vs <- mapMList (fmap Identity . runTermE env) ts
+    pure $ uncurryList_ runIdentity (sem f) vs
+
+checkPredE ::
```

Comment:
Maybe add a test that this function has "the same" result as the existing `checkPred` function?

Reply:
Alternatively do it on the spec level, but you get the point.

Reply:
Ok so I write a test that compares "the Same" for conformsToSpec and conformsToSpecE, it classifies the results by what form og Specification is used, and if the test succeeds because both functions conform, or both function do not conform. A sample run looks like this.
```
Set Integer [✔]
      +++ OK, passed 1000 tests:
      36.5% "SuspendedSpec both fail to conform"
      28.9% "TypeSpec both fail to conform"
      20.1% "MemberSpec both fail to conform"
       3.6% "SuspendedSpec both conform"
       3.5% "TrueSpec both conform"
       2.2% "ErrorSpec both fail to conform"
       2.1% "(ExplainSpec SuspendedSpec) both fail to conform"
       1.1% "(ExplainSpec MemberSpec) both fail to conform"
       0.7% "(ExplainSpec TypeSpec) both fail to conform"
       0.4% "(ExplainSpec SuspendedSpec) both conform"
       0.3% "TypeSpec both conform"
       0.2% "MemberSpec both conform"
       0.1% "(ExplainSpec (ExplainSpec ErrorSpec)) both fail to conform"
       0.1% "(ExplainSpec (ExplainSpec SuspendedSpec)) both fail to conform"
       0.1% "(ExplainSpec TrueSpec) both conform"
       0.1% "(ExplainSpec TypeSpec) both conform"
```